### PR TITLE
Re-structured Doc

### PR DESCRIPTION
**Pull request by tuoping**
Modified the index structures.

### CODE REVIEW

**Comment by tuoping** [comment]
The latest commit changed LICENSE in the main directory to LICENSE.rst and linked it to index.rst.
A new file credits.rst, with all the contributor's GITHUB usernames are listed. Zhang ZiXuan as the designer of logo is listed under "other credits".
Side bar is reconstructed into four sections.
API is modified and recomplied.
Use-DeePMD-kit is renames as getting-started.
Pair-style-deepmd-command is merged into getting-started, under "Run MD with Lammps".
Lammps, i-PI and ASE are incorporated under a same entry "Run MD".
Install.rst is seperated to 2 parts: install_easy.rst and install_source.rst.

File: doc/credits.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+* DiracMD
+* dongdawn
+* felix5572
+* jameswind
+* jiaweile
+* jxxiaoshaoye
+* kevinwenminion
+* KuangYu
+* lqs0609
+* mohanchen
```
Comment (marian-code) [inline doc/credits.rst:20]:
```suggestion
* marian-code
* mohanchen
```

**Review by marian-code — CHANGES_REQUESTED** [review]
Could you please add my name to the list? I do not want to create new PR just for that.

File: doc/getting-started.md [hunk 2]
```diff
@@ -271,14 +331,14 @@ kspace_modify	gewald 0.45
...
-## Run path-integral MD with i-PI
+### Run path-integral MD with i-PI
 The i-PI works in a client-server model. The i-PI provides the server for integrating the replica positions of atoms, while the DeePMD-kit provides a client named `dp_ipi` that computes the interactions (including energy, force and virial). The server and client communicates via the Unix domain socket or the Internet socket. The client can be started by
 ```bash
 $ dp_ipi water.json
 ```
 It is noted that multiple instances of the client is allow for computing, in parallel, the interactions of multiple replica of the path-integral MD.
 
-`water.json` is the parameter file for the client `dp_ipi`, and [an example](./examples/ipi/water.json) is provided:
+`water.json` is the parameter file for the client `dp_ipi`, and [an example](https://github.com/tuoping/deepmd-kit/blob/devel/examples/water/ipi/water.json) is provided:
```
Comment (amcadmus) [inline doc/getting-started.md:341]: please avoid hard coding

File: doc/getting-started.md [hunk 3]
```diff
@@ -138,7 +151,7 @@ dp train input.json
 
 If enbled in json/yaml input file DeePMD-kit will create log files which can be
 used to analyze training procedure with Tensorboard. For a short tutorial
-please read this [document](doc/tensorboard.md).
+please read this [document](tensorboard.html).
```
Comment (amcadmus) [inline doc/getting-started.md:154]: Is it a valid link? I do not see tensorboard.html in our source

**Comment by amcadmus** [comment]
There is a mixed use of "-" and "_" in the documentation file names. I would prefer that we only use "-".

File: doc/credits.rst (hunk 1 above)
Comment (tuoping) [reply doc/credits.rst:20]: Of course. Sorry I missed you.

File: doc/getting-started.md (hunk 3 above)
Comment (tuoping) [reply doc/getting-started.md:137]: This link is valid. All the html files are compiled and under directory _build.

Comment (tuoping) [reply doc/getting-started.md:137]: I changed it back anyway.

File: doc/getting-started.md (hunk 2 above)
Comment (tuoping) [reply doc/getting-started.md]: Files in a parent dir can't be linked. I deleted it in the next commit (4.21 15:00).

File: doc/application-examples.md [hunk 4]
```diff
@@ -6,3 +6,8 @@
 
 ## MD on different hardware platforms
 
+There is a scheme to generate doc html from python scripts provided in the ../examples/ directory, so that each case in doc has a counter part in ../examples.
+
+This scheme is implemented with sphinx-gallery.
+
+Do we want to use this ???
```
Comment (amcadmus) [inline doc/application-examples.md:13]: Why do we need this file? what does the text mean ?

File: doc/credits.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+* haidi
+* Jiequn Han
+* Yixiao Chen
+* bwang-ecnu
+* hlyang
+* Linfeng Zhang
+* deepmodeling
+* iProzd
+* njzjz
+* wsyxbcl
```
Comment (amcadmus) [inline doc/credits.rst:30]: It is messy to use a mixed list of real names and github IDs. Could you please clean it up?
Some of the item are duplicated, Marian Rynic and marian-code is the same person. Duo and iprozd are the same person.

File: doc/getting-started.md [hunk 6]
```diff
@@ -82,16 +79,16 @@ One can use the a convenient tool `dpdata` to convert data directly from the out
 A model has two parts, a descriptor that maps atomic configuration to a set of symmetry invariant features, and a fitting net that takes descriptor as input and predicts the atomic contribution to the target physical property.
 
 DeePMD-kit implements the following descriptors:
-1. [`se_e2_a`](train-se-e2-a.md#descriptor): DeepPot-SE constructed from all information (both angular and radial) of atomic configurations. The embedding takes the distance between atoms as input.
-2. [`se_e2_r`](train-se-e2-r.md): DeepPot-SE constructed from radial information of atomic configurations. The embedding takes the distance between atoms as input.
-3. [`se_e3`](train-se-e3.md): DeepPot-SE constructed from all information (both angular and radial) of atomic configurations. The embedding takes angles between two neighboring atoms as input.
+1. <a rel="subsection" href="train-se-e2-a.html#descriptor">`se_e2_a`</a>: DeepPot-SE constructed from all information (both angular and radial) of atomic configurations. The embedding takes the distance between atoms as input.
```
Comment (amcadmus) [inline doc/getting-started.md:82]: It breaks the link in markdown.

File: doc/index.rst [hunk 7]
```diff
@@ -3,28 +3,48 @@
...
    contain the root `toctree` directive.
 
+==========================
 DeePMD-kit's documentation
-======================================
+==========================
+
+DeePMD-kit is a package written in Python/C++, designed to minimize the effort required to build deep learning based model of interatomic potential energy and force field and to perform molecular dynamics (MD). This brings new hopes to addressing the accuracy-versus-efficiency dilemma in molecular simulations. Applications of DeePMD-kit span from finite molecules to extended systems and from metallic systems to chemically bonded systems.
+
+.. Important:: The project DeePMD-kit is licensed under GNU LGPLv3.0. If you use this code in any future publications, please cite this using *Han Wang, Linfeng Zhang, Jiequn Han, and Weinan E. "DeePMD-kit: A deep learning package for many-body potential energy representation and molecular dynamics." Computer Physics Communications 228 (2018): 178-184.*
```
Comment (amcadmus) [inline doc/index.rst:12]: Provide the link to LGPLv3

File: doc/install-hardware-platforms.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+
+# Installation on different hardware platforms
```
Comment (amcadmus) [inline doc/install-hardware-platforms.md:2]: Why do we need this file?

File: doc/whatsnew.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-# Novel Auxiliary Options
+# What's New
```
Comment (amcadmus) [inline doc/whatsnew.md:1]: This file is messy, we need to clean it up

File: doc/conf.py [hunk 10]
```diff
@@ -30,6 +31,7 @@
 extensions = [
     'recommonmark',
     "sphinx_rtd_theme",
+	'sphinx_markdown_tables',
```
Comment (njzjz) [inline doc/conf.py:34]:
```suggestion
    'sphinx_markdown_tables',
```
Do not use any <tab> in Python files.

File: doc/getting-started.md [hunk 11]
```diff
@@ -324,3 +368,5 @@ dyn = BFGS(water)
 dyn.run(fmax=1e-6)
 print(water.get_positions())
 ```
+[DP]:https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.120.143001
+[DP-SE]:https://arxiv.org/abs/1805.09003
```
Comment (njzjz) [inline doc/getting-started.md:372]:
```suggestion
[DP-SE]:https://dl.acm.org/doi/10.5555/3327345.3327356
```
Use a publish version instead?

File: doc/api.rst [hunk 12]
```diff
@@ -1,91 +1,107 @@
...
    :members:
    :undoc-members:
 
-.. automodule:: deepmd.DescrptSeA
+.. automodule:: deepmd.descriptor.se_ar
    :members:
    :undoc-members:
 
-.. automodule:: deepmd.DescrptSeAR
+.. automodule:: deepmd.descriptor.se_r
```
Comment (njzjz) [inline doc/api.rst:48]: I think some other packages are missing here?

**Review by njzjz — CHANGES_REQUESTED** [review]

File: doc/application-examples.md (hunk 4 above)
Comment (tuoping) [reply doc/application-examples.md:13]: This is about toy model. If we are going to add something like "application examples", there is a scheme called sphinx-gallery, where we put toy models in example file, and generate demos automatically to display in doc.

Comment (tuoping) [reply doc/application-examples.md:13]: We don't have toy model yet. So this is just a suggestion to put in bank.

File: doc/credits.rst (hunk 5 above)
Comment (tuoping) [reply doc/credits.rst:30]: Sorry. It's fixed now.

File: doc/getting-started.md (hunk 6 above)
Comment (tuoping) [reply doc/getting-started.md:82]: Yes. The html links are supposed to be built from markdown automatically by "sphinx-build". But only head of an external file can be automatically linked. Sphinx-build cannot create the  `rel="subsection"` text. 
So, I can either change these links to html style and abandon markdown links, or I'll have to modify the sphinx source code, and upload a new compiler to readthedocs.org to make everything work.

File: doc/index.rst (hunk 7 above)
Comment (tuoping) [reply doc/index.rst:12]: ok.

File: doc/install-hardware-platforms.md (hunk 8 above)
Comment (tuoping) [reply doc/install-hardware-platforms.md:2]: I don't remember anymore. I'll remove it.

File: doc/whatsnew.md (hunk 9 above)
Comment (tuoping) [reply doc/whatsnew.md:1]: First of all, do we need it? If we do, do we need both section " Interpolation with tabulated pair potentials" and "Type embedding"?

File: doc/conf.py (hunk 10 above)
Comment (tuoping) [reply doc/conf.py:34]: Ok.

File: doc/getting-started.md (hunk 11 above)
Comment (tuoping) [reply doc/getting-started.md:372]: Ok.

File: doc/api.rst (hunk 12 above)
Comment (tuoping) [reply doc/api.rst:48]: Indeed. Thank you.

File: doc/getting-started.md (hunk 6 above)
Comment (tuoping) [reply doc/getting-started.md]: It's working now. Please take a look.

File: requirements.txt [hunk 13]
```diff
@@ -4,3 +4,4 @@ pyyaml
 dargs >= 0.2.2
 tqdm
 typing_extensions
+sphinx_markdown_tables
```
Comment (njzjz) [inline requirements.txt:7]: Move this package to the following line, as we do not need it unless building the document.
https://github.com/deepmodeling/deepmd-kit/blob/40c6ae2a70b8aa6eb2c3e64dd928d63efa7e4083/setup.py#L110

File: doc/requirements.txt [hunk 14]
```diff
@@ -1 +1,2 @@
 .[docs,cpu]
+myst-parser
```
Comment (njzjz) [inline doc/requirements.txt:2]: Also this one

File: requirements.txt (hunk 13 above)
Comment (tuoping) [reply requirements.txt:7]: ok

File: doc/requirements.txt (hunk 14 above)
Comment (tuoping) [reply doc/requirements.txt:2]: ok

File: doc/getting-started.md (hunk 6 above)
Comment (amcadmus) [reply doc/getting-started.md]: Great!

File: doc/whatsnew.md (hunk 9 above)
Comment (amcadmus) [reply doc/whatsnew.md:1]: We need to put what's new in v2.0, but not in the current form.

File: doc/application-examples.md (hunk 4 above)
Comment (amcadmus) [reply doc/application-examples.md:13]: see examples/* and doc/train-*.md for latest updates on the examples.

**Review by amcadmus — CHANGES_REQUESTED** [review]
Some links are broken because you have changed the file names, for example
https://github.com/tuoping/deepmd-kit/blob/devel/doc/use-deepmd-kit.md
Please check all the links we have in the doc and make sure all of them are working.

File: doc/credits.rst (hunk 5 above)
Comment (amcadmus) [reply doc/credits.rst:27]: I suggest showing the contributors in an alphabetical order

File: doc/getting-started.md [hunk 15]
```diff
@@ -271,14 +315,14 @@ kspace_modify	gewald 0.45
...
-## Run path-integral MD with i-PI
+### Run path-integral MD with i-PI
 The i-PI works in a client-server model. The i-PI provides the server for integrating the replica positions of atoms, while the DeePMD-kit provides a client named `dp_ipi` that computes the interactions (including energy, force and virial). The server and client communicates via the Unix domain socket or the Internet socket. The client can be started by
 ```bash
 $ dp_ipi water.json
 ```
 It is noted that multiple instances of the client is allow for computing, in parallel, the interactions of multiple replica of the path-integral MD.
 
-`water.json` is the parameter file for the client `dp_ipi`, and [an example](./examples/ipi/water.json) is provided:
+`water.json` is the parameter file for the client `dp_ipi`, and an example is provided:
```
Comment (amcadmus) [inline doc/getting-started.md:325]: why removing the link?

Comment (amcadmus) [reply doc/getting-started.md:325]: The reason is that in html we cannot find the file, am I right?

File: doc/credits.rst (hunk 5 above)
Comment (tuoping) [reply doc/credits.rst:27]: Of course, Sorry.

File: doc/whatsnew.md (hunk 9 above)
Comment (tuoping) [reply doc/whatsnew.md:1]: OK.

File: doc/getting-started.md (hunk 15 above)
Comment (tuoping) [reply doc/getting-started.md:325]: Yes.

**Comment by tuoping** [comment]
> Some links are broken because you have changed the file names, for example
> https://github.com/tuoping/deepmd-kit/blob/devel/doc/use-deepmd-kit.md
> Please check all the links we have in the doc and make sure all of them are working.

Thanks. I changed use-deepmd-kit.md to getting-started.md, so there isn't a use-deepmd-kit.md anymore. And I checked all the links. They should be fine now.

**Comment by amcadmus** [comment]
I see ` doc/whatsnew.md` is copied from the "hightlight in v2.0" part of `README.md`. It is not recommended to copy the code to different files. One may create a link to ` doc/whatsnew.md` in `README.md`

**Comment by tuoping** [comment]
> I see ` doc/whatsnew.md` is copied from the "hightlight in v2.0" part of `README.md`. It is not recommended to copy the code to different files. One may create a link to ` doc/whatsnew.md` in `README.md`

OK.

File: README.md [hunk 16]
```diff
@@ -19,16 +19,7 @@ DeePMD-kit is a package written in Python/C++, designed to minimize the effort r
...
-
-* [Model compression](doc/use-deepmd-kit.md#compress-a-model). Accelerate the efficiency of model inference for 4-15 times.
-* [New descriptors](doc/use-deepmd-kit.md#write-the-input-script). Including [`se_e2_r`](doc/train-se-e2-r.md) and [`se_e3`](doc/train-se-e3.md).
-* [Hybridization of descriptors](doc/train-hybrid.md). Hybrid descriptor constructed from concatenation of several descriptors.
-* Atom type embedding.
-* Training and inference the dipole (vector) and polarizability (matrix).
-* Split of training and validation dataset.
-* Optimized training on GPUs. 
-
+<iframe src="doc/_build/html/highlights.html" />
```
Comment (njzjz) [inline README.md:22]: I don't think this will work in GitHub Flavored Markdown

Comment (tuoping) [reply README.md]: You are right.

File: README.md [hunk 17]
```diff
@@ -19,16 +19,7 @@ DeePMD-kit is a package written in Python/C++, designed to minimize the effort r
...
-
-* [Model compression](doc/use-deepmd-kit.md#compress-a-model). Accelerate the efficiency of model inference for 4-15 times.
-* [New descriptors](doc/use-deepmd-kit.md#write-the-input-script). Including [`se_e2_r`](doc/train-se-e2-r.md) and [`se_e3`](doc/train-se-e3.md).
-* [Hybridization of descriptors](doc/train-hybrid.md). Hybrid descriptor constructed from concatenation of several descriptors.
-* Atom type embedding.
-* Training and inference the dipole (vector) and polarizability (matrix).
-* Split of training and validation dataset.
-* Optimized training on GPUs. 
-
+<iframe src="doc/_build/html/highlights.html"></iframe>
```
Comment (amcadmus) [inline README.md:22]: It does not work in markdown.

Comment (tuoping) [reply README.md]: Yes, I removed it.

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+## What you can contribute
+
+You can either make a code contribution, help improve our document or offer help to other users. Your help is always appreciated. Come and have fun!
+
+### Code contribution
+You can start from any one of the following items to help improve deepmd-kit
+
+- Smash a bug
+- Implement a feature or add a patch, whatever you think deepmd-kit is missing
+- Browse the deepmd-kit [roadmap](doc/roadmap.md) and pick a bug you want to smash or a feature you want to implement
```
Comment (amcadmus) [inline CONTRIBUTING.md:14]: We do not put the bugs and feature requests in `doc/roadmap.md`, we put them in [issues](https://github.com/deepmodeling/deepmd-kit/issues)

**Review by amcadmus — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 18 above)
Comment (tuoping) [reply CONTRIBUTING.md:14]: ok

File: README.md [hunk 19]
```diff
@@ -19,17 +19,15 @@ DeePMD-kit is a package written in Python/C++, designed to minimize the effort r
...
 
-## Highlights in DeePMD-kit v2.0
-
-* [Model compression](doc/use-deepmd-kit.md#compress-a-model). Accelerate the efficiency of model inference for 4-15 times.
-* [New descriptors](doc/use-deepmd-kit.md#write-the-input-script). Including [`se_e2_r`](doc/train-se-e2-r.md) and [`se_e3`](doc/train-se-e3.md).
-* [Hybridization of descriptors](doc/train-hybrid.md). Hybrid descriptor constructed from concatenation of several descriptors.
+# Heighlights in DeePMD-kit v2.0
+* [Model compression](getting-started.md#compress-a-model). Accelerate the efficiency of model inference for 4-15 times.
+* [New descriptors](getting-started.md#write-the-input-script). Including [`se_e2_r`](train-se-e2-r.md) and [`se_e3`](train-se-e3.md).
+* [Hybridization of descriptors](train-hybrid.md). Hybrid descriptor constructed from concatenation of several descriptors.
```
[the remark this answers is not in the document]
Comment (tuoping) [reply README.md:25]: sorry

**Comment by tuoping** [comment]
Added two functions in conf.py to detect files in folders troubleshooting/ and development/, and append them in troubleshooting/index.md and development/index.md automatically. 
Sorry, forgot to mention this in COMMIT_MSG.

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+
+### Step 1: Fork the repository
+
+1. Visit the project: <https://github.com/deepmodeling/deepmd-kit>
+2. Click the **Fork** button on the top right and wait it to finish.
+
+### Step 2: Clone the forked repository to local storage and set configurations
+
+1. Clone your own repo, not the public repo (from deepmodeling) ! And change the branch to devel.
+    ```
```
Comment (njzjz) [inline CONTRIBUTING.md:54]:
```suggestion
    ```bash
```
Please assign the language for all codes, such as `bash` and `python`.

File: README.md [hunk 21]
```diff
@@ -19,17 +19,15 @@ DeePMD-kit is a package written in Python/C++, designed to minimize the effort r
 
 For more information, check the [documentation](https://deepmd.readthedocs.io/).
 
-## Highlights in DeePMD-kit v2.0
-
-* [Model compression](doc/use-deepmd-kit.md#compress-a-model). Accelerate the efficiency of model inference for 4-15 times.
-* [New descriptors](doc/use-deepmd-kit.md#write-the-input-script). Including [`se_e2_r`](doc/train-se-e2-r.md) and [`se_e3`](doc/train-se-e3.md).
+# Heighlights in DeePMD-kit v2.0
```
Comment (njzjz) [inline README.md:22]: What is `heighlights`? I think `highlights` is the correct word.

File: CONTRIBUTING.md (hunk 20 above)
Comment (tuoping) [reply CONTRIBUTING.md:54]: Of course, Sorry.

File: doc/troubleshooting/index.rst [hunk 22]
```diff
@@ -0,0 +1,12 @@
...
+
+In consequence of various differences of computers or systems, problems may occur. Some common circumstances are listed as follows. 
+If other unexpected problems occur, you're welcome to contact us for help.
+
+.. toctree::
+   :maxdepth: 2
+   
+   installation.md
+   md-version-compatibility.md
+   model-compatability.md
```
[the remark this answers is not in the document]
Comment (tuoping) [reply doc/troubleshooting/index.rst]: I changed doc/troubleshooting/index.rst to doc/troubleshooting/index.md, they should work in markdown environment now.

File: README.md (hunk 21 above)
Comment (tuoping) [reply README.md:22]: Ops, sorry.

File: README.md [hunk 23]
```diff
@@ -5,7 +5,7 @@
 
 # Table of contents
 - [About DeePMD-kit](#about-deepmd-kit)
- 	- [Highlights in v2.0](#highlights-in-deepmd-kit-v2.0)
+ 	- [Highlights in v2.0](#heighlights-in-deepmd-kit-v2.0)
```
Comment (njzjz) [inline README.md:8]:
```suggestion
 	- [Highlights in v2.0](#highlights-in-deepmd-kit-v2.0)
```

**Review by njzjz — APPROVED** [review]

File: doc/development/index.md [hunk 24]
```diff
@@ -0,0 +1,4 @@
+# Developer Guide
+   
+- [coding-conventions](coding-conventions.rst)
+- [api](api.rst)
```
Comment (njzjz) [inline doc/development/index.md:4]: These two links use the filename instead of the title?

Comment (tuoping) [reply doc/development/index.md:4]: I'll change that.

**Pull request merged**